packages/core/plugin/src/PluginAPI.d.ts

```diff
@@ -0,0 +1,47 @@
+import type {
+  Transformer as TransformerOpts,
+  Resolver as ResolverOpts,
+  Bundler as BundlerOpts,
+  Namer as NamerOpts,
+  Runtime as RuntimeOpts,
+  Packager as PackagerOpts,
+  Optimizer as OptimizerOpts,
+  Reporter as ReporterOpts,
+  Validator as ValidatorOpts,
+} from '@parcel/types';
+  
+export declare class Transformer<T> {
```

Comment:
Did this file manually because flow allows the type parameter on the constructor rather than the class but TS doesn't. Wanted to avoid the type param on the class in Flow otherwise you have to explicitly type the config on every plugin rather than it being inferred. This is because flow doesn't infer the types of exports in type first mode and forces you to type it explicitly (which is really annoying).